server/main.go

```diff
@@ -72,6 +72,12 @@ func (m *Main) Run(ctx context.Context) error {
 		if err := trillian.RegisterTrillianAdminHandlerFromEndpoint(ctx, mux, m.RPCEndpoint, opts); err != nil {
 			return err
 		}
+		if err := trillian.RegisterTrillianLogHandlerFromEndpoint(ctx, mux, m.RPCEndpoint, opts); err != nil {
```

Comment:
We shouldn't register Log / Map gateway handlers here, as we don't know here which are really available. (We register Admin because it's present in all servers.)

Instead, do this by taking advantage of RegisterHandlerFunc. See links for [logserver](https://github.com/google/trillian/blob/master/server/trillian_log_server/main.go#L111) and [mapserver](https://github.com/google/trillian/blob/master/server/trillian_map_server/main.go#L90).